include/nbl/ui/IInputEvent.h

```diff
@@ -0,0 +1,68 @@
+#ifndef __NBL_UI_I_INPUT_EVENT_H_INCLUDED__
+#define __NBL_UI_I_INPUT_EVENT_H_INCLUDED__
+#include <chrono>
+
+namespace nbl::ui
+{
+class IWindow;
+
+struct IEvent
```

Comment:
its not an interface, its a simple struct, should be prefixed with `S`, `SEvent`... but also the header name should match the struct name, so `SInputEvent.h` and `SInputEventBase` (also nice because `SEvent` name collides with a bunch of engines)